underfs/s3/src/main/java/alluxio/underfs/s3/S3UnderFileSystem.java

```diff
@@ -459,7 +460,22 @@ private boolean isFolder(String key) {
       // If no exception is thrown, the key exists as a folder
       return true;
     } catch (ServiceException s) {
-      return false;
+      // It is possible that the folder has not been encoded as a _$folder$ file
+      try {
+        String dir = stripPrefixIfPresent(key);
+        String dirPrefix = dir.endsWith(PATH_SEPARATOR) ? dir : dir + PATH_SEPARATOR;
+        // Check if anything begins with <folder_path>/
+        S3Object[] objs = mClient.listObjects(mBucketName, dirPrefix, "");
+        // If there are, this is a folder and we can create the necessary metadata
+        if (objs.length > 0) {
+          mkdirsInternal(dir);
+          return true;
+        } else {
+          return false;
+        }
+      } catch (ServiceException s2) {
+        return false;
```

Comment:
I think it would be noisy since we expect this code path when the path is not a folder or doesn't exist.